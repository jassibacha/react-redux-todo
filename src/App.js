import './App.css';
import TodoList from './TodoList';

function App() {
    return (
        <div className="App container mt-5">
            <h1 className="text-center mb-4">To-Do List</h1>
            <TodoList />
        </div>
    );
}

export default App;
