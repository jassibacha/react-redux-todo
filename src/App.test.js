import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

describe('App', () => {
    let store;

    beforeEach(() => {
        store = mockStore({
            todos: [],
        });
    });

    test('renders without crashing', () => {
        render(
            <Provider store={store}>
                <App />
            </Provider>
        );
    });

    test('renders To-Do List header', () => {
        render(
            <Provider store={store}>
                <App />
            </Provider>
        );
        const headerElement = screen.getByText(/To-Do List/i);
        expect(headerElement).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <App />
            </Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('submits form in TodoForm and adds new todo', () => {
        render(
            <Provider store={store}>
                <App />
            </Provider>
        );

        // Simulate filling out the form
        fireEvent.change(screen.getByPlaceholderText('Enter a new task'), {
            target: { value: 'New Task' },
        });

        // Simulate form submission
        fireEvent.click(screen.getByText('Add'));

        // Check if the 'ADD_TODO' action was dispatched
        const actions = store.getActions();
        expect(actions[0].type).toBe('ADD_TODO');
        expect(actions[0].payload).toEqual({
            task: 'New Task',
            id: expect.any(String), // Since ID is generated dynamically
        });
    });

    // Additional tests...
});
