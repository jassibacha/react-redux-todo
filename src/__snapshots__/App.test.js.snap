// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App container mt-5"
  >
    <h1
      class="text-center mb-4"
    >
      To-Do List
    </h1>
    <div
      class="Todos"
    >
      <div
        class="card bg-light mb-3"
      >
        <div
          class="card-body"
        >
          <form
            class="TodoForm row"
          >
            <div
              class="col-10"
            >
              <input
                class="form-control"
                id="task"
                name="task"
                placeholder="Enter a new task"
                type="text"
                value=""
              />
            </div>
            <div
              class="col-2"
            >
              <button
                class="btn btn-primary w-100"
                type="submit"
              >
                Add
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="mt-3"
      />
    </div>
  </div>
</DocumentFragment>
`;
